### PR TITLE
MSDKUI-1839: Adapt tests to location injection

### PR DESCRIPTION
**Pull request by ants86**
- Updated tests to work with new solution
- Enabled guidance tests for CI with the exception
of speed/speed limit related tests
- Removed mock location

Signed-off-by: Ants Peetsalu <43877908+ants86@users.noreply.github.com>

### CODE REVIEW

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt [hunk 1]
```diff
@@ -202,22 +208,36 @@ open class CoreActions {
...
      */
-    fun provideMockLocation(mockLocationData: MockLocationData): CoreActions {
-        mockLocationData.run {
-            if (isMocked) mockLocation!!.setMockLocation(testPlace)
-        }
-        return this
+    fun setCurrentLocation(off: Boolean = false, locationData: MapData = GEO_POINT_0) {
+        //Map engine must be initialized before we can inject coordinates.
+        waitForMap()
+        val appPositioningManager = AppPositioningManager.getInstance()
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: we can make syntax in more kotlin way..something like this
AppPositioningManager.getInstance().apply {
if(off) reset() 
else customLocation = GeoCoordinate(locationData.lat, locationData.lng)
}

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt [hunk 2]
```diff
@@ -202,22 +208,36 @@ open class CoreActions {
...
-     * Stop mocking location service
+     * Wait for map engine to be initialized
      */
-    fun stopMockLocation(mockLocationData: MockLocationData): CoreActions {
-        mockLocationData.run {
-            if (isMocked) mockLocation!!.stopMocking()
-        }
-        return this
+    private fun waitForMap() {
+        val mapIdlingResource = CountingIdlingResource("MapEngineIdlingResource")
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: here u can use with keyword
with(CountingIdlingResource("MapEngineIdlingResource")) {
val idlingRegistry = IdlingRegistry.getInstance()
 idlingRegistry.register(this)
if(.....) {
decrement()
idlingRegistry.unregister(this)


}

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt [hunk 3]
```diff
@@ -362,8 +346,8 @@ class GuidanceFlowTests : TestBase<SplashActivity>(SplashActivity::class.java) {
             onGuidanceDashBoardPullLine.perform(click())
             DriveNavigationActions.waitGuidanceDashBoardCollapse()
             DriveNavigationMatchers.checkGuidanceDashBoardExpanded(false)
-            // Expand dashboard by swipe and check
-            onView(withId(R.id.collapsed_view)).perform(CoreActions().swipeTwiceUp())
+            // Expand dashboard by click and check
+            onView(withId(R.id.collapsed_view)).perform(click())
```
Comment (annvar) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt]: I think you cannot change swipe by click. Those are test steps to test expand/collapse by swipe.

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt [hunk 4]
```diff
@@ -320,8 +307,7 @@ class GuidanceFlowTests : TestBase<SplashActivity>(SplashActivity::class.java) {
             GuidanceActions.waitForGuidanceNextManeuverChanged(streetName)
             // Recheck next-next maneuver panel elements
             DriveNavigationMatchers.checkNextManeuverPanelElementsDisplayed()
-
-            // Stop the guidance with "X" button
+            // Stop guidance
```
Comment (annvar) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt]: you changed here the description but there are other tests with the same description ;)

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt [hunk 5]
```diff
@@ -202,22 +208,36 @@ open class CoreActions {
...
+        val appPositioningManager = AppPositioningManager.getInstance()
+        val lat = locationData.lat
+        val long = locationData.lng
+            if (!off) {
+                appPositioningManager.run {
+                    customLocation = GeoCoordinate(lat, long)
+                }
+            } else {
+                appPositioningManager.reset()
+            }
```
Comment (annvar) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: All good, but just curious to know if "If" is with one more level deeper(tabs) by intention or not :)

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt (hunk 2 above)
Comment (ants86) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: Done

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt (hunk 3 above)
Comment (ants86) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt]: Done

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt (hunk 1 above)
Comment (ants86) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: Done

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt (hunk 5 above)
Comment (ants86) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: a mistake, removed :)

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt (hunk 4 above)
Comment (ants86) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/guidance/flow/GuidanceFlowTests.kt]: Done

**Pull request merged**